src/Commands/Generators/UserModelGenerator.php

```diff
@@ -0,0 +1,78 @@
+<?php
+
+declare(strict_types=1);
+
+namespace CodeIgniter\Shield\Commands\Generators;
+
+use CodeIgniter\CLI\BaseCommand;
+use CodeIgniter\CLI\CLI;
+use CodeIgniter\CLI\GeneratorTrait;
+
+/**
+ * Generates a custom user model file.
+ */
+class UserModelGenerator extends BaseCommand
+{
+    use GeneratorTrait;
+
+    /**
+     * @var string
+     */
+    protected $group = 'Shield';
+
+    /**
+     * @var string
+     */
+    protected $name = 'shield:model';
+
+    /**
+     * @var string
+     */
+    protected $description = 'Generate a new UserModel file.';
+
+    /**
+     * @var string
+     */
+    protected $usage = 'shield:model [<name>] [options]';
+
+    /**
+     * @var array<string, string>
+     */
+    protected $arguments = [
+        'name' => 'The model class name. If not provided, this will default to `UserModel`.',
+    ];
+
+    /**
+     * @var array<string, string>
+     */
+    protected $options = [
+        '--namespace' => 'Set root namespace. Default: "APP_NAMESPACE".',
+        '--suffix'    => 'Append the component title to the class name (e.g. User => UserModel).',
+        '--force'     => 'Force overwrite existing file.',
+    ];
+
+    /**
+     * Actually execute the command.
+     */
+    public function run(array $params): void
+    {
+        $this->component = 'Model';
+        $this->directory = 'Models';
+        $this->template  = 'usermodel.tpl.php';
+
+        $this->classNameLang = 'CLI.generator.className.model';
+        $this->setHasClassName(false);
+
+        $class = $params[0] ?? CLI::getSegment(2) ?? 'UserModel';
+
+        if (in_array(strtolower($class), ['shielduser', 'shieldusermodel'], true)) {
```

Comment:
Why do we need `shielduser`? Cannot generate class `ShieldUser`?

Reply:
How about this?

```php
        $classToCheck = CLI::getOption('suffix') ? $class . 'Model' : $class;
        if (strtolower($classToCheck) === 'shieldusermodel') {
            CLI::error('Cannot use `ShieldUserModel` as class name as this conflicts with the parent class.', 'light_gray', 'red');

            return; // @TODO when CI4 is at v4.3+, change this to `return 1;` to signify failing exit
        }
```

Reply:
Updated. See https://github.com/codeigniter4/shield/pull/558/commits/7e5c4a2069cf570f7e08b934a0f7633c17f7c12e

Reply:
Okay, thanks!